client/ts/src/auth/auth.ts

```diff
@@ -84,4 +85,18 @@ export class Client {
     };
     return mw;
   }
+
+  async register(username: string, password: string): Promise<user.Payload> {
+    const [res, err] = await this.client.send(
+      REGISTER_ENDPOINT,
+      { username: username, password: password },
+      insecureCredentialsZ,
+      tokenResponseZ,
+    );
+
+    if (err != null) {
```

Comment:
There is a utility function in freighter called `sendRequired` us you should use instead of using the `err != null` checks